components/camel-sql/src/main/java/org/apache/camel/component/sql/SqlConsumer.java

```diff
@@ -101,8 +101,8 @@ public DefaultSqlEndpoint getEndpoint() {
     }
 
     @Override
-    protected void doStart() throws Exception {
-        super.doStart();
+    protected void doInit() throws Exception {
+        super.doInit();
```

Comment:
This should be doStart as SqlHelper.resolveQuery can potentially load the SQL template from file system. See the SQL docs